caml_z.c

```diff
@@ -1758,8 +1758,8 @@ CAMLprim value ml_z_gcd(value arg1, value arg2)
     mp_size_t sz, pos1, pos2, limb1, limb2, bit1, bit2, pos, limb, bit, i;
     Z_DECL(arg1); Z_DECL(arg2);
     Z_ARG(arg1);  Z_ARG(arg2);
-    if (!size_arg1) r = arg2;
-    else if (!size_arg2) r = arg1;
+    if (!size_arg1) r = sign_arg2 ? ml_z_neg(arg2) : arg2;
```

Comment:
What about `ml_z_abs(arg2)`?

Reply:
It's probably a little faster in the (usual) case where the arguments are positive to avoid the call to ml_z_abs.
This also makes me realize that ml_z_abs is not optimal when its argument is positive: it returns a copy!

Reply:
OK, as you prefer.  My feeling is that the case where one of the arguments is 0 is pretty uncommon to begin with, so speed doesn't matter much.

Re `ml_z_abs` not optimal: yes, it would be nice to avoid the copy when nonnegative.